internal/initwd/module_install.go

```diff
@@ -245,6 +245,17 @@ func (i *ModuleInstaller) installLocalModule(req *earlyconfig.ModuleRequest, key
 	// filesystem at all because the parent already wrote
 	// the files we need, and so we just load up what's already here.
 	newDir := filepath.Join(parentRecord.Dir, req.SourceAddr)
+
+	// it is possible that the local directory is a symlink
+	newDir, err := filepath.EvalSymlinks(newDir)
+	if err != nil {
+		diags = diags.Append(tfdiags.Sourceless(
+			tfdiags.Error,
+			"Unreadable module directory",
```

Comment:
To describe what's happening here, it's "Unable to evaluate directory symlink" perhaps? That's redundant with the "Unable" detail, but I like symlink in the main error

Reply:
That's a much better phrasing - I'm going to use it for the detail, though, not the summary. 
My intent was to keep the summary for this error the same as the one below, since they are similar situations.

Now that I look at this again I'm also going to move this bit of code below the preceding `log` output; the `[TRACE]` info is relevant!